tinygrad/codegen/uops.py

```diff
@@ -99,21 +102,23 @@ def divides(self, v):
       if self.arg is BinaryOps.MUL: return any(x.divides(v) for x in self.src)
     return False # generic false if we aren't sure
   @functools.cached_property
-  def vmin(self) -> UOp: return x if (x:=self._min_max[0]) is not None else self.const(dtypes.min(cast(DType, self.dtype)))
+  def vmin(self) -> UOp: return x if (x:=self._min_max[0]) is not None else self.const(dtypes.min(cast(DType, self.dtype))).vmin
   @functools.cached_property
-  def vmax(self) -> UOp: return x if (x:=self._min_max[1]) is not None else self.const(dtypes.max(cast(DType, self.dtype)))
+  def vmax(self) -> UOp: return x if (x:=self._min_max[1]) is not None else self.const(dtypes.max(cast(DType, self.dtype))).vmax
   @functools.cached_property
   def _min_max(self) -> Tuple[Optional[UOp], Optional[UOp]]:
     # NOTE: returned UOp is assumed to be CONST
     if self.op in (UOps.DEFINE_VAR, UOps.RANGE): return self.src[0], self.src[1] if isinstance(self.src[1].arg, int) else None
     # TODO: UOps.SPECIAL is UOps.DEFINE_VAR
     if self.op is UOps.SPECIAL: return self.const(0), self.const(self.arg[1]-1) if isinstance(self.arg[1], int) else None
     if self.op is UOps.CONST: return self, self
+    if self.op is UOps.VECTORIZE and all(x.op == UOps.CONST for x in self.src):
+      return min(self.src, key=lambda x: x.arg), max(self.src, key=lambda x: x.arg)
```

Comment:
hmm, I think this is extra complexity. you always have the same const across a vectorize.
Can it just be self.const(dtypes.max(self.dtype.scalar())) in the above?

Reply:
> you always have the same const across a vectorize.
Not true. I've seen different value consts being vectorized in test_real_world.py. I was surprised by this and I recorded it in my notes while I was hacking at the PTX renderer (see the unhighlighted white text above the diff):
![Pasted Graphic 3](https://github.com/user-attachments/assets/58100559-cb53-47a7-a5fa-883ba1e06e8e)

I agree that it is extra complexity though -- I definitely need to cleanup the vmin & vmax logic. Will try to have a fix out shortly, but I need to understand the upstream behavior a little better. chenyu's generic max folding was a sore point for my rebasing haha

Reply:
ohh right you can have a const folded valid mixed with invalid.

Reply:
I think I've got the vmin/vmax stuff reasonably clean, if the extra `sconst` def is acceptable. The `sconst` def's purpose is to save lines and satisfy mypy, otherwise we'd have casts everywhere.

I checked the process_replay and it's consistent with what we were seeing before the cleanup.